code.cpp

```diff
@@ -0,0 +1,194 @@
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <random>
+#include <vector>
+#include <climits>
+
+bool createFileWithRandomNumbers(const std::string &fileName, const int numbersCount, const int maxNumberValue) {
+    std::ofstream file(fileName);
+    if (!file.is_open()) {
+        std::cerr << "cant open file: " << fileName << std::endl;
+        return false;
+    }
+
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> dist(0, maxNumberValue);
+
+    for (int i = 0; i < numbersCount; ++i) {
+        file << dist(gen) << " ";
+        if (!file) {
+            std::cerr << "cant write into the file: " << fileName << std::endl;
+            file.close();
+            return false;
+        }
+    }
+
+    file.close();
+    return true;
+}
+
+bool isFileContainsSortedArray(const std::string &fileName) {
+    int valueNow = 0, valuePrev = INT_MIN;
+    std::string ch;
+    std::ifstream file(fileName);
+    while (!file.eof()) {
+        ch = "";
+        file >> ch;
+        if (ch != "") {
+            valueNow = std::stoi(ch);
+            if (valueNow < valuePrev)
+                return false;
+            valuePrev = valueNow;
+        }
+    }
+    return true;
+}
+
+void split(const std::string &filename, const std::string &fileSplit1, const std::string &fileSplit2) {
+    std::ifstream file(filename);
+    std::ofstream file_1(fileSplit1);
+    std::ofstream file_2(fileSplit2);
+
+    int value;
+    file >> value;
+
+    while (!file.eof()) {
+        file_1 << value << " ";
+        file >> value;
+        if (!file.eof()) {
```

Comment:
Кроме `eof`, могут быть ошибки чтения. Надёжнее использовать `good`:
```suggestion
    while (file.good()) {
        file_1 << value << " ";
        file >> value;
        if (file.good()) {
```